models/dataset.py

```diff
@@ -0,0 +1,184 @@
+
+    """
+    for i, data in enumerate(dataset):
+        min_data = torch.min(data)
+        max_data = torch.max(data)
+        if max_data != min_data:
+            raise ZeroDivisionError
+        dataset[i] = (data - min_data) / (max_data - min_data)
+    return dataset
+
+
+def organize_dataset(dataset, batch_size, frac_val):
+    """
+    Parameters
+    ----------
+    dataset : tensor of images.
+    batch_size : int batch_size.
+    frac_val : ratio between validation and training datasets.
+
+    Returns
+    -------
+    trainset : tensor of training images.
+    testset : tensor of test images.
+    trainloader : tensor ready to be used by the NN for training images.
+    testloader : tensor ready to be used by the NN for test images.
+
+    """
+    N = len(dataset)
+    n = int(N*frac_val)
+    trainset, testset = random_split(dataset, [N-n, n])
+    trainloader = DataLoader(
+        trainset, batch_size=batch_size, shuffle=True, **KWARGS)
+    testloader = DataLoader(
+        testset, batch_size=batch_size, shuffle=False, **KWARGS)
+    return trainset, testset, trainloader, testloader
+
+
+PATHS = {}
+PATHS['simulated2D'] = "cryo_sim_128x128.npy"
+PATHS['real_2D30'] = "class2D_30_sort.h5"
+PATHS['real_2D39'] = "class2D_39_sort.h5"
+PATHS['real_2D93'] = "class2D_93_sort.h5"
+PATHS['refine3D'] = "refine3D_180x180_sort.h5"
+PATHS['class3D9090'] = "class3D_90x90_sort.h5"
+PATHS['simulated3D'] = "ConcatSimulated.npy"
+PATHS['simulated3DNoise'] = "cryo_sim_128x128.npy"
+PATHS['real3D'] = 'data.hdf5'
+PATHS['4Points'] = "4Points.npy"
+PATHS['4Points1'] = "4Points1.npy"
+PATHS['4Points3D'] = "3Dimages.npy"
+
+
+SHAPES = {}
+SHAPES['simulated3D'] = (1, 320, 320)
+SHAPES['simulated2D'] = (1, 128, 128)
+SHAPES['4Points'] = (1, 128, 128)
+SHAPES['4Points1'] = (1, 64, 64)
+SHAPES['4Points3D'] = (64, 64, 64)
+
+CONSTANTES = {}
+CONSTANTES['img_shape'] = (1, 128, 128)
+CONSTANTES['3D'] = False
+CONSTANTES['with_sigmoid'] = True
+CONSTANTES['OUT_CHANNELS'] = [32, 64]
+CONSTANTES['conv_dim'] = len(CONSTANTES['img_shape'][1:])
+CONSTANTES['dataset_name'] = PATHS['simulated2D']
+CONSTANTES['ENC_KS'] = 4
+CONSTANTES['ENC_STR'] = 2
+CONSTANTES['ENC_PAD'] = 1
+CONSTANTES['ENC_DIL'] = 1
+CONSTANTES['ENC_C'] = 1
+CONSTANTES['DEC_KS'] = 3
+CONSTANTES['DEC_STR'] = 1
+CONSTANTES['DEC_PAD'] = 0
+CONSTANTES['DEC_DIL'] = 1
+CONSTANTES['DEC_C'] = 1
+CONSTANTES['DIS_KS'] = 4
+CONSTANTES['DIS_STR'] = 2
+CONSTANTES['DIS_PAD'] = 1
+CONSTANTES['DIS_DIL'] = 1
+CONSTANTES['DIS_C'] = 1
+CONSTANTES['regularizations'] = ('kullbackleibler')
+CONSTANTES['class_2d'] = 39
+CONSTANTES['weights_init'] = 'xavier'
+CONSTANTES['nn_type'] = 'conv'
+CONSTANTES['beta1'] = 0.9
+CONSTANTES['beta2'] = 0.999
+CONSTANTES['frac_val'] = 0.2
+CONSTANTES['BCE'] = True
+CONSTANTES['DATA_DIM'] = functools.reduce(
+    (lambda x, y: x * y), CONSTANTES['img_shape'])
+CONSTANTES['reconstructions'] = ('bce_on_intensities', 'adversarial')
+CONSTANTES['SKIP_Z'] = False
```

Comment:
Since we have a lot of CONSTANTS here, let's organize them in yml files:
`datasets_config.yml` and `nn_config.yml` and import the yml file as a dictionary: https://stackabuse.com/reading-and-writing-yaml-to-a-file-in-python

Reply:
i think it's good to have acces to this constants, so you  are able to change them really quickly